libvast/vast/format/json.hpp

```diff
@@ -53,33 +56,53 @@ caf::error add(table_slice_builder& builder, const vast::json::object& xs,
 
 /// @relates reader
 struct default_selector {
-  caf::optional<record_type> operator()(const vast::json::object&) {
-    return layout;
+  caf::optional<record_type> operator()(const vast::json::object& obj) const {
+    std::vector<std::string> cache_entry;
+    auto build_cache_entry
+      = [&cache_entry](const std::string& prefix, const vast::json&) {
+          cache_entry.emplace_back(prefix);
+        };
+    each_field(vast::json{obj}, build_cache_entry);
```

Comment:
I think we should operate on `obj` *directly*. Otherwise we construct a new object for every invocation of `each_field`.